### PR TITLE
Fix uses decrement and add local sdb check on kv v2

### LINKED ISSUE #57705: [BUG] SDB Vault Config Fails on "uses"
**Description**
When attempting to configure a vault connection and sdb setup, salt is simply unable to get appropriate values from vault in any capacity, complaining about the sdb vault script and a "uses" KeyError.

**Setup**
Salt Version 3001
Vault 1.4.2 running locally in dev mode

**Steps to Reproduce the behavior**
1. Follow this guide to bootstrap salt w/ vault sdb:
https://help.saltstack.com/hc/en-us/articles/360041140451-Quick-Guide-to-Vault-Integration

### Error Message
```
/etc/salt# salt-run sdb.get 'sdb://localvault/salt/data/test/foo'
[ERROR   ] Failed to read secret! KeyError: 'uses'
Exception occurred in runner sdb.get: Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/salt/sdb/vault.py", line 97, in get
    response = __utils__["vault.make_request"]("GET", url, profile)
  File "/usr/lib/python3/dist-packages/salt/utils/vault.py", line 326, in make_request
    **args
  File "/usr/lib/python3/dist-packages/salt/utils/vault.py", line 342, in make_request
    connection["uses"] -= 1
KeyError: 'uses'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/salt/client/mixins.py", line 397, in low
    data["return"] = func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/salt/runners/sdb.py", line 26, in get
    return salt.utils.sdb.sdb_get(uri, __opts__, __utils__)
  File "/usr/lib/python3/dist-packages/salt/utils/sdb.py", line 46, in sdb_get
    return loaded_db[fun](query, profile=profile)
  File "/usr/lib/python3/dist-packages/salt/sdb/vault.py", line 113, in get
    raise salt.exceptions.CommandExecutionError(e)
salt.exceptions.CommandExecutionError: 'uses
```
### Master Config:
```
vault:
  url: http://127.0.0.1:8200
  auth:
    method: token
    token: ROOT_TOKEN
  policies:
    - saltmaster

localvault:
  driver: vault
```
### Vault Evidence
```
# vault kv get salt/test
====== Metadata ======
Key              Value
---              -----
created_time     2020-06-17T19:35:11.776643342Z
deletion_time    n/a
destroyed        false
version          1

=== Data ===
Key    Value
---    -----
foo    bar
```

**Expected behavior**
Salt with a vault sdb is able to successfully pull a secret from vault (either kv1 or kv2)

**Versions Report**
```
# salt -V
Salt Version:
           Salt: 3001

Dependency Versions:
           cffi: 1.14.0
       cherrypy: Not Installed
       dateutil: 2.8.1
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
         Jinja2: 2.11.2
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: 2.20
       pycrypto: 3.9.7
   pycryptodome: 3.9.7
         pygit2: Not Installed
         Python: 3.5.2 (default, Apr 16 2020, 17:47:17)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 17.1.2
          smmap: 3.0.4
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.4

System Versions:
           dist: ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1101-aws
         system: Linux
        version: Ubuntu 16.04 xenial
```
</details>

**Additional context**
Upgraded to 3001 today after it was released due to the desire to connect to a v2 kv store on vault. Even attempting vault kv1 with a root token produces the same problem and a complete inability to get secrets out.

Would not be surprised if it was my configuration, but I'd assume the quickstart guide would still be relevant.

### PR DESCRIPTION
**Pull request by mchugh19**
### What does this PR do?
Fixes situation where `salt-call --local` or configuration with role_id calls to vault do not support uses key.

### What issues does this PR fix or reference?
Fixes: #57705

### Previous Behavior
```
salt.exceptions.CommandExecutionError: 'uses
```

### New Behavior
Commands succeed as expected.
```
# salt-call --local sdb.get 'sdb://localvault/kv-v2/test/test_sdb/foo'
local:
    bar
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by mchugh19** [comment]
Hey all. Since #57705 is an awful regression, this should probably be ported to 3001.1

File: tests/integration/sdb/test_vault.py [hunk 1]
```diff
@@ -241,6 +252,24 @@ def test_sdb_kv2(self):
         )
         self.assertEqual(get_output, "bar")
 
+    @flaky
```
Comment (DmitryKuzmenko) [inline tests/integration/sdb/test_vault.py:255]: You're marked this test as `flaky`. Is the test initially designed to be unstable?

**Comment by DmitryKuzmenko** [comment]
@mchugh19 test failures look related. Could you address it please?

File: tests/integration/sdb/test_vault.py (hunk 1 above)
Comment (mchugh19) [reply tests/integration/sdb/test_vault.py:255]: It's a copy of the surrounding functions. The tests launch a vault docker image. It generally works for me, but the whole class is marked slow and flaky, so I'm guessing there were issues with it in the past.

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by DmitryKuzmenko** [comment]
re-run full all

**Comment by DmitryKuzmenko** [comment]
@mchugh19 thank you for update on this. Tests passed but your test is a slowtest so I've restarted tests in full mode to check the slowtests results.

**Comment by DmitryKuzmenko** [comment]
re-run full all

**Comment by DmitryKuzmenko** [comment]
re-run full all

**Comment by DmitryKuzmenko** [comment]
re-run full all

**Comment by mchugh19** [comment]
@DmitryKuzmenko the windows failures seem to be timeouts and unrelated.

**Comment by mchugh19** [comment]
re-run full all

**Comment by DmitryKuzmenko** [comment]
@mchugh19 I know but we have to follow the procedure. The windows tests are fixed now. So I've updated your PR and will re-run full tests soon.

**Comment by DmitryKuzmenko** [comment]
re-run full windows

**Comment by himbob** [comment]
We needed this fix in production fast so and pulled down your proposed changes and tested live.  All our **sdb://myvault/salt/data** commands work now in 3001 (patched-kinda-by-you ).  I know its not good practice, but it works!!!  TY!

**Comment by DmitryKuzmenko** [comment]
Now it's done. Let's wait for merge.

**Review by dwoz — APPROVED** [review]

**Comment by richardh538** [comment]
In #57989 an additional context is given that is not solved (maybe because the issue was closed as a duplicate).
I have manually patched the python files for the "uses" problem.

Now the error is: "Non-listing operations on the root of a K/V v2 mount are not supported."

[DEBUG ] Starting new HTTP connection (1): 127.0.0.1:8200
[DEBUG ] http://127.0.0.1:8200 "GET /v1/salt/data HTTP/1.1" 404 232
[ERROR ] Error from vault: {"request_id":"af546575-64bf-8254-ace3-b102df2c5efe","lease_id":"","renewable":false,"lease_duration":0,"data":null,"wrap_info":null,"warnings":["Non-listing operations on the root of a K/V v2 mount are not supported."],"auth":null}
[DEBUG ] LazyLoaded local_cache.prep_jid
[DEBUG ] Sending event: tag = salt/run/20201112084715709358/ret; data = {'fun': 'runner.sdb.get', 'jid': '20201112084715709358', 'user': 'salt', 'fun_args': ['sdb://myvault/salt/data/user1'], '_stamp': '2020-11-12T08:47:15.928509', 'return': None, 'success': True}
[DEBUG ] LazyLoaded json.output
null
[INFO ] Runner completed: 20201112084715709358
[DEBUG ] Closing IPCMessageClient instance
[DEBUG ] Runner return: None

**Comment by lfanew** [comment]
> In #57989 an additional context is given that is not solved (maybe because the issue was closed as a duplicate).
> I have manually patched the python files for the "uses" problem.
> 
> Now the error is: "Non-listing operations on the root of a K/V v2 mount are not supported."
> 
> # salt-run sdb.get 'sdb://myvault/salt/data/user1' --out=json -l debug
> ...

I too ran into this when following SaltStack's [Quick Guide for Vault Integration](https://help.saltstack.com/hc/en-us/articles/360041140451-Quick-Guide-to-Vault-Integration). I believe the problem lies within the guide.

* It instructs you to retrieve an entire secret (user1) with sdb.vault.get
    * `salt-run sdb.get 'sdb://myvault/salt/data/user1'`
    * Running this will give you the error stated above - **Non-listing operations on the root of a K/V v2 mount are not supported**.
* Looking at the guide, `salt/data/user1` was a secret defined with two KV fields
    * `vault kv put salt/user1 password=p4ssw0rd desc="test user"` - so this secrets contains two KV pairs.
*  Now, looking at the documentation for sdb.vault.get we see the function documented as:
    * `salt.sdb.vault.get(key, profile=None)` - **Get a value from the vault service**
* Notice how sdb.vault.get takes a _key_ as an argument, not a secret. With that in mind let's run something that works.
    * `salt-run sdb.get 'sdb://myvault/salt/data/user1/password`
        * [OUT] **p4ssw0rd**
    * `salt-run sdb.get 'sdb://myvault/salt/data/user1/desc`
        * [OUT] **test user**

**Comment by s0undt3ch** [comment]
@lfanew would you mind opening an issue? It looks like we have a doc fix to do, at least.

/cc @ScriptAutomate

**Pull request merged**